vendor/blockchain_contracts/src/lib.rs

```diff
@@ -0,0 +1,66 @@
+#![warn(unused_extern_crates, missing_debug_implementations, rust_2018_idioms)]
+#![deny(unsafe_code)]
+
+#[macro_use]
+extern crate serde;
+
+use std::{cmp::Ordering, fmt::Display};
+
+pub mod ethereum;
+pub mod rfc003;
+
+#[derive(Debug, Eq)]
+pub struct Offset {
+    data: String,
+    start: usize,
+    end: usize,
+    length: usize,
+}
+
+impl Display for Offset {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> Result<(), std::fmt::Error> {
+        write!(
+            f,
+            "{}\t{}\t\t{}\t{}",
+            self.data, self.start, self.end, self.length
+        )
+    }
+}
+
+pub fn format_table(mut offsets: Vec<Offset>) -> String {
+    let mut res = String::from("Data\t\t\t\tStart\tEnd\tLength");
```

Comment:
I think it's a fair idea. @LLFourn had some opinion on the matter so I tagged him in your other comment.

Reply:
I think we should have start and end and just say start is **included** while end is **excluded**, rather than first byte, last byte so you can translate it directly into a rust (and many languages) range: eg. 42..62.